From 00d2ae7f94d7c89d6adbd134a8e09af3b7aa152c Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Mon, 4 Oct 2021 18:48:30 -0700
Subject: [PATCH 80/85] tools/meson: update to 0.59.2

Update install procedure based on upstream feedback. Normally, meson is
to be installed with pip. But as pip is not mandated by the build
system, it cannot be used. Upstream provides a nice script to pack meson
automatically.

Moved src/ to files/. No need to copy to BUILD_DIR.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 include/meson.mk                                 | 3 +--
 tools/meson/Makefile                             | 9 ++++++---
 tools/meson/{src => files}/openwrt-cross.txt.in  | 0
 tools/meson/{src => files}/openwrt-native.txt.in | 0
 4 files changed, 7 insertions(+), 5 deletions(-)
 rename tools/meson/{src => files}/openwrt-cross.txt.in (100%)
 rename tools/meson/{src => files}/openwrt-native.txt.in (100%)

diff --git a/include/meson.mk b/include/meson.mk
index 01d0ce0280..9862f0e662 100644
--- a/include/meson.mk
+++ b/include/meson.mk
@@ -19,7 +19,6 @@
 #
 # Host packages are built in the same fashion, just use these vars instead:
 #
-# HOST_BUILD_DEPENDS:=meson/host
 # MESON_HOST_ARGS+=-Dfoo -Dbar=baz
 # MESON_HOST_VARS+=FOO=bar
 
@@ -57,7 +56,7 @@ MESON_CPU:="$(CPU_TYPE)$(if $(CPU_SUBTYPE),+$(CPU_SUBTYPE))"
 endif
 
 define Meson
-	$(2) $(STAGING_DIR_HOST)/bin/$(PYTHON) $(MESON_DIR)/meson.py $(1)
+	$(2) $(STAGING_DIR_HOST)/bin/meson $(1)
 endef
 
 define Meson/CreateNativeFile
diff --git a/tools/meson/Makefile b/tools/meson/Makefile
index bae89654a5..0cf0efba47 100644
--- a/tools/meson/Makefile
+++ b/tools/meson/Makefile
@@ -1,11 +1,11 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=meson
-PKG_VERSION:=0.59.1
+PKG_VERSION:=0.59.2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://github.com/mesonbuild/meson/releases/download/$(PKG_VERSION)
-PKG_HASH:=db586a451650d46bbe10984a87b79d9bcdc1caebf38d8e189f8848f8d502356d
+PKG_HASH:=13dee549a7ba758b7e33ce7719f28d1d337a98d10d378a4779ccc996f5a2fc49
 
 PKG_MAINTAINER:=Andre Heider <a.heider@gmail.com>
 PKG_LICENSE:=Apache-2.0
@@ -20,8 +20,11 @@ define Host/Compile
 endef
 
 define Host/Install
+	$(INSTALL_DIR) $(STAGING_DIR_HOST)/bin
+	$(HOST_BUILD_DIR)/packaging/create_zipapp.py $(HOST_BUILD_DIR) --interpreter $(STAGING_DIR_HOST)/bin/$(PYTHON) --outfile $(STAGING_DIR_HOST)/bin/meson
 	$(INSTALL_DIR) $(STAGING_DIR_HOST)/lib/meson
-	$(CP) $(HOST_BUILD_DIR)/* $(STAGING_DIR_HOST)/lib/meson/
+	$(INSTALL_CONF) files/openwrt-cross.txt.in $(STAGING_DIR_HOST)/lib/meson/
+	$(INSTALL_CONF) files/openwrt-native.txt.in $(STAGING_DIR_HOST)/lib/meson/
 endef
 
 define Host/Clean
diff --git a/tools/meson/src/openwrt-cross.txt.in b/tools/meson/files/openwrt-cross.txt.in
similarity index 100%
rename from tools/meson/src/openwrt-cross.txt.in
rename to tools/meson/files/openwrt-cross.txt.in
diff --git a/tools/meson/src/openwrt-native.txt.in b/tools/meson/files/openwrt-native.txt.in
similarity index 100%
rename from tools/meson/src/openwrt-native.txt.in
rename to tools/meson/files/openwrt-native.txt.in
-- 
2.35.1

